src/views/SignTransactionLedger.vue

```diff
@@ -385,27 +384,30 @@ export default class SignTransactionLedger extends Vue {
             return this.cashlink ? this.cashlink.message : null;
         }
 
-        const request = this.request as ParsedSignTransactionRequest | ParsedCheckoutRequest;
-        if (!request.data || request.data.byteLength === 0) {
-            return null;
-        }
-
+        let data;
         let flags;
         if (this.request.kind === RequestType.SIGN_TRANSACTION) {
             const signTransactionRequest = this.request as ParsedSignTransactionRequest;
+            data = signTransactionRequest.data;
             flags = signTransactionRequest.flags;
         } else {
+            data = this.checkoutPaymentOptions!.protocolSpecific.extraData;
             flags = this.checkoutPaymentOptions!.protocolSpecific.flags;
         }
```

Comment:
Using destructuring here would be nice as is done for other assignments in this class.